centreon/src/Core/ResourceAccess/Infrastructure/Repository/Providers/HostCategoryRepositoryProvider.php

```diff
@@ -0,0 +1,110 @@
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\ResourceAccess\Infrastructure\Repository\Providers;
+
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
+use Core\Common\Infrastructure\Repository\RepositoryTrait;
+use Core\HostCategory\Application\Repository\ReadHostCategoryRepositoryInterface;
+use Core\ResourceAccess\Application\Exception\RuleException;
+use Core\ResourceAccess\Application\Repository\DatasetRepositoryProviderInterface;
+use Core\ResourceAccess\Domain\Model\DatasetFilter\Providers\HostCategoryFilterType;
+
+final class HostCategoryRepositoryProvider extends AbstractRepositoryDRB implements DatasetRepositoryProviderInterface
+{
+    use LoggerTrait, RepositoryTrait;
+
+    /**
+     * @param DatabaseConnection $db
+     * @param ReadHostCategoryRepositoryInterface $repository
+     */
+    public function __construct(
+        DatabaseConnection $db,
+        private readonly ReadHostCategoryRepositoryInterface $repository
+    ) {
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function isValidFor(string $type): bool
+    {
+        return HostCategoryFilterType::TYPE_NAME === $type;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function areResourcesValid(array $resourceIds): array
+    {
+        $validIds = $this->repository->exist($resourceIds);
+        if ([] !== ($invalidIds = array_diff($resourceIds, $validIds))) {
+            throw RuleException::idsDoNotExist(HostCategoryFilterType::TYPE_NAME, $invalidIds);
+        }
+
+        return $validIds;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function linkResourcesToDataset(int $ruleId, int $datasetId, array $resourceIds): void
+    {
+        $this->debug(
+            'Add relation between host categories and dataset for rule',
+            ['rule_id' => $ruleId, 'dataset_id' => $datasetId, 'host_category_ids' => $resourceIds]
+        );
+
+        if ([] === $resourceIds) {
+            return;
+        }
+
+        $bindValues = [];
+        $subValues = [];
+        foreach ($resourceIds as $index => $hostCategoryId) {
+            $bindValues[":host_category_id_{$index}"] = $hostCategoryId;
+            $subValues[] = "(:host_category_id_{$index}, :datasetId)";
+        }
+
+        $subQueries = implode(', ', $subValues);
+
+        $request = $this->translateDbName(
+            <<<SQL
+                    INSERT INTO `:db`.acl_resources_hc_relations (hc_id, acl_res_id)
```

Comment:
⚠️ POSSIBLY UNWANTED SIDE SIDE EFFECT
categories and severities ACLs work as "filter", with means that by default you have access to all the categories/severities (aka. when the table acl_resources_hc_relations is empty for a given acl_res_id)
By adding an element in the table, you are effectively **cutting access** to all other categories/severities

Reply:
That is indeed the goal. However in Resource Access Management there will not be such thing as exclude / filter resources

Reply:
But I see what you mean